core-metrics/src/main/java/org/apache/kylin/metrics/lib/impl/BlockingReservoir.java

```diff
@@ -60,14 +60,15 @@ public BlockingReservoir(int minReportSize, int maxReportSize) {
     }
 
     public BlockingReservoir(int minReportSize, int maxReportSize, int maxReportTime) {
-        this(minReportSize, maxReportSize, maxReportSize, MAX_QUEUE_SIZE);
+        this(minReportSize, maxReportSize, maxReportTime, MAX_QUEUE_SIZE);
     }
 
     public BlockingReservoir(int minReportSize, int maxReportSize, int maxReportTime, int maxQueueSize) {
         Preconditions.checkArgument(minReportSize > 0, "minReportSize should be larger than 0");
         Preconditions.checkArgument(maxReportSize >= minReportSize,
                 "maxReportSize should not be less than minBatchSize");
         Preconditions.checkArgument(maxReportTime > 0, "maxReportTime should be larger than 0");
+        Preconditions.checkArgument(maxQueueSize > 0, "maxQueueSize should be larger than 0");
```

Comment:
How about making it possible to use unbounded LinkedBlockingQueue like following?

this.recordsQueue = maxQueueSize <= 0 ? new LinkedBlockingQueue<>() : new LinkedBlockingQueue<>(maxQueueSize);

Reply:
Good advice!